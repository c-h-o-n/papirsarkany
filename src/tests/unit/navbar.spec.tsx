import { render } from "@testing-library/react";
import { afterEach, expect, test } from "vitest";

import Navbar from "~/components/navbar";
import twConfig from "~/lib/tailwind-theme.preval";
import { viMockMatchMedia } from "~/mocks/matchMedia.mock";

const mediaQueries = {
  desktop: `(min-width: ${twConfig.screens.md})`,
};

afterEach(() => {
  viMockMatchMedia({
    media: mediaQueries.desktop,
    matches: false,
  });
});

test(`should render desktop-navbar on screens larger then ${twConfig.screens.md}`, () => {
  viMockMatchMedia({
    media: mediaQueries.desktop,
    matches: true,
  });

  const { container } = render(<Navbar />);

  expect(container).toMatchSnapshot();
});

test(`should render mobile-navbar on screens smaller then ${twConfig.screens.md}`, () => {
  viMockMatchMedia({
    media: mediaQueries.desktop,
    matches: false,
  });
  const { container } = render(<Navbar />);
  expect(container).toMatchSnapshot();
});
