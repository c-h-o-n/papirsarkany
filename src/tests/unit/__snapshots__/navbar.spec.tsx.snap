// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`should render desktop-navbar on screens larger then 768px 1`] = `
<div>
  <div
    class="sticky top-0 z-40"
  >
    <div
      class="d-navbar border-b-4 border-black bg-white"
    >
      <div
        class="d-navbar-start gap-1"
      >
        <a
          aria-label="home"
          class="d-btn d-btn-ghost py-1 text-xl normal-case hover:!bg-sky-200"
          href="/#hello"
        >
          <svg
            class="h-full"
            viewBox="0 0 512 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M32.337,512L0,479.663c14.539-14.539,15.794-25.841,17.695-42.946 c2.195-19.76,4.928-44.352,30.811-70.232s50.473-28.614,70.232-30.811c17.105-1.901,28.407-3.157,42.946-17.695l43.115-43.115 l32.337,32.337l-43.115,43.115c-25.882,25.882-50.473,28.614-70.232,30.811c-17.105,1.901-28.407,3.157-42.946,17.695 c-14.539,14.538-15.794,25.841-17.695,42.946C60.95,461.527,58.219,486.118,32.337,512z"
              style="fill: #DFDFE1;"
            />
            <polygon
              points="512,0 231.747,0 177.853,334.147 512,280.253 "
              style="fill: #7AB9E8;"
            />
            <polygon
              points="512,0 177.853,334.147 512,280.253 "
              style="fill: #579ADF;"
            />
            <polygon
              points="404.21,140.126 512,32.337 512,0 479.663,0 371.874,107.79 264.084,0 231.747,0  227.256,27.846 339.537,140.126 184.071,295.591 177.853,334.147 216.407,327.929 371.874,172.463 484.154,284.744 512,280.253  512,247.916 "
              style="fill: #4A82CA;"
            />
          </svg>
        </a>
        <ul
          class="d-menu d-menu-horizontal flex-nowrap gap-2 text-base font-extrabold"
        >
          <li
            class="text-sm lg:text-base"
          >
            <a
              href="/#vallalkozas"
            >
              A vállalkozásról
            </a>
          </li>
          <li
            class="text-sm lg:text-base"
          >
            <a
              href="/#sarkany-keszites"
            >
              Sárkány készítés
            </a>
          </li>
          <li
            class="text-sm lg:text-base"
          >
            <a
              href="/#elerhetoseg"
            >
              Elérhetőség
            </a>
          </li>
        </ul>
      </div>
      <div
        class="d-navbar-end"
      >
        <ul
          class="d-menu d-menu-horizontal gap-2 text-base font-extrabold"
        >
          <li
            class="text-sm lg:text-base"
          >
            <a
              href="/sarkanyok"
            >
              Sárkányok
            </a>
          </li>
          <li
            class="text-sm lg:text-base"
          >
            <a
              href="/anyagok"
            >
              Anyagok
            </a>
          </li>
        </ul>
        <div
          class="flex-none"
        >
          <a
            class="d-btn-circle"
            data-pw-e2e="cart-menu-item"
            href="/kosar"
          >
            <div
              class="p-0"
            >
              <div
                class="d-btn d-btn-circle d-btn-ghost"
              >
                <div
                  class="d-indicator"
                >
                  <svg
                    class="h-5 w-5"
                    fill="none"
                    stroke="currentColor"
                    stroke-linecap="round"
                    stroke-linejoin="round"
                    stroke-width="2"
                    viewBox="0 0 24 24"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M3 3h2l.4 2M7 13h10l4-8H5.4M7 13L5.4 5M7 13l-2.293 2.293c-.63.63-.184 1.707.707 1.707H17m0 0a2 2 0 100 4 2 2 0 000-4zm-8 2a2 2 0 11-4 0 2 2 0 014 0z"
                    />
                  </svg>
                  <span
                    class="d-badge d-indicator-item d-badge-sm"
                    data-testid="cart-menu-item-total-count"
                  >
                    0
                  </span>
                </div>
              </div>
            </div>
          </a>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`should render mobile-navbar on screens smaller then 768px 1`] = `
<div>
  <div
    class="sticky top-0 z-40"
  >
    <div
      class="d-navbar border-b-4 border-black bg-white"
    >
      <div
        class="d-navbar-start gap-1"
      >
        <div
          class="d-dropdown"
        >
          <div
            aria-label="mobile navigation menu"
            class="d-btn d-btn-ghost lg:hidden"
            role="button"
            tabindex="0"
          >
            <svg
              class="h-5 w-5"
              stroke="currentColor"
              viewBox="0 0 24 24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M4 6h16M4 12h8m-8 6h16"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="2"
                style="stroke-width: 2;"
              />
            </svg>
          </div>
          <ul
            class="d-menu d-dropdown-content z-[1] mt-5 w-52 rounded-box bg-base-100 p-2 shadow"
            data-pw-e2e="hamburger-menu-content"
            tabindex="0"
          >
            <li
              class="text-sm lg:text-base"
            >
              <a
                href="/sarkanyok"
              >
                Sárkányok
              </a>
            </li>
            <li
              class="text-sm lg:text-base"
            >
              <a
                href="/anyagok"
              >
                Anyagok
              </a>
            </li>
            <div
              class="d-divider m-0"
            />
            <li
              class="text-sm lg:text-base"
            >
              <a
                href="/#vallalkozas"
              >
                A vállalkozásról
              </a>
            </li>
            <li
              class="text-sm lg:text-base"
            >
              <a
                href="/#sarkany-keszites"
              >
                Sárkány készítés
              </a>
            </li>
            <li
              class="text-sm lg:text-base"
            >
              <a
                href="/#elerhetoseg"
              >
                Elérhetőség
              </a>
            </li>
          </ul>
        </div>
        <div
          class="flex-none"
        >
          <a
            class="d-btn-circle"
            data-pw-e2e="cart-menu-item"
            href="/kosar"
          >
            <div
              class="p-0"
            >
              <div
                class="d-btn d-btn-circle d-btn-ghost"
              >
                <div
                  class="d-indicator"
                >
                  <svg
                    class="h-5 w-5"
                    fill="none"
                    stroke="currentColor"
                    stroke-linecap="round"
                    stroke-linejoin="round"
                    stroke-width="2"
                    viewBox="0 0 24 24"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M3 3h2l.4 2M7 13h10l4-8H5.4M7 13L5.4 5M7 13l-2.293 2.293c-.63.63-.184 1.707.707 1.707H17m0 0a2 2 0 100 4 2 2 0 000-4zm-8 2a2 2 0 11-4 0 2 2 0 014 0z"
                    />
                  </svg>
                  <span
                    class="d-badge d-indicator-item d-badge-sm"
                    data-testid="cart-menu-item-total-count"
                  >
                    0
                  </span>
                </div>
              </div>
            </div>
          </a>
        </div>
      </div>
      <div
        class="d-navbar-end"
      >
        <a
          class="d-btn d-btn-ghost normal-case min-[320px]:text-xl"
          href="/"
        >
          papirsarkany.hu
        </a>
      </div>
    </div>
  </div>
</div>
`;
